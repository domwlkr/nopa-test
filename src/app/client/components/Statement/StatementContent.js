import React, {PropTypes} from 'react';
import { Button } from '../index';
import AccountDetails from './AccountDetails';
import TransactionsTable from './TransactionsTable';
import * as Paths from '../../constants/paths';

const StatementContent = (props) => {
  return (
    <div className="main-content">
      <AccountDetails accountDetails={props.accountDetails} bank={props.bank} />
      <TransactionsTable transactions={props.transactions}
        loading={props.loading} />
      <Button onClick={props.showMore} className="button">Show more</Button>
    </div>
  );
};

StatementContent.propTypes = {
  accountDetails: PropTypes.object,
  loading: PropTypes.bool,
  bank: PropTypes.object,
  showMore: PropTypes.func,
  transactions: PropTypes.array
};

export default StatementContent;