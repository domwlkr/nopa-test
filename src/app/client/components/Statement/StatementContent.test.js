import React from 'react';
import StatementContent from './StatementContent';
import renderer from 'react-test-renderer';

const bank = {name: 'Barclays', logo: 'test-file-stub'};
const details = {surname: 'Walker', accountNumber: '90909090', sortCode: '220099'};
const loading = false;
const showMore = jest.fn();
const transactions = [];

test('Renders correctly', () => {
  const statementContent = renderer.create(
    <StatementContent accountDetails={details} loading={loading} bank={bank} showMore={showMore} transactions={transactions} />
  ).toJSON();

  expect(statementContent).toMatchSnapshot();
});