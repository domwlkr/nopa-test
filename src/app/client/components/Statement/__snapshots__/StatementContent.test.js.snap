// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="main-content"
>
  <div
    className="account-details"
  >
    <div
      className="account-details__left"
    >
      <span>
        Barclays
      </span>
      <span>
        Walker
      </span>
    </div>
    <div
      className="account-details__right"
    >
      <span>
        Current account
      </span>
      <span>
        90909090
      </span>
      <span>
        220099
      </span>
    </div>
  </div>
  <div
    className="table-wrap"
  >
    <p>
      Your transactions for the last 30 days
    </p>
    <div
      className="transactions-table"
    />
  </div>
  <a
    className="button"
    onClick={[Function]}
    style={Object {}}
  >
    Show more
  </a>
</div>
`;
